modelforge/tests/test_dataset.py

```diff
@@ -288,11 +304,36 @@ def test_numpy_dataset_assignment(dataset):
     """
     Test if the numpy_dataset attribute is correctly assigned after processing or loading.
     """
-    from modelforge.dataset.transformation import default_transformation
 
     factory = DatasetFactory()
     data = dataset(for_unit_testing=True)
-    factory._load_or_process_data(data, None, default_transformation)
+    factory._load_or_process_data(data)
 
     assert hasattr(data, "numpy_data")
     assert isinstance(data.numpy_data, np.lib.npyio.NpzFile)
+
+
+def test_self_energy():
+
+    from modelforge.dataset.dataset import TorchDataModule
+
+    # test the self energy calculation on the QM9 dataset
+    from modelforge.dataset.qm9 import QM9Dataset
+    from modelforge.dataset.utils import FirstComeFirstServeSplittingStrategy
+
+    data = QM9Dataset(for_unit_testing=True)
+    dataset = TorchDataModule(
+        data, batch_size=32, split=FirstComeFirstServeSplittingStrategy()
+    )
+
+    # self energy is calculated and removed in prepare_data if `remove_self_energies` is True
+    dataset.prepare_data(remove_self_energies=True, normalize=True)
+
+    assert dataset.dataset_statistics
+    self_energies = dataset.dataset_statistics["self_energies"]
+    # only 4 elements present in the reduced QM9 dataset
+    assert len(self_energies) == 4
```

Comment:
I am not completely sure why these are off actually, they matched the last time I checked this. I will open an issue about this.

Reply:
I wouldn't say these are really all that off.  All but hydrogen have less than a 1% difference.  Hydrogen actually has the smallest numerial difference (delta), but because it is nearly 2 orders of magnitude smaller, it just seems more significant.  

```
element |  DFT              | regressed         | delta         | % diff
H       |   -1313.466862    |   -1585.320503    |  271.8536413  |  20.69741151
C       |  -99366.70746     |  -99963.18962     |  596.4821665  |   0.600283719
N       | -143309.938       | -143744.2422      |  434.3042769  |   0.303052449
O       | -197082.0672      | -197504.4753      |  422.4081034  |   0.21433107
F       | -261811.5456      | -262199.961       |  388.415491   |   0.148356899
```

But since the regressed values are all consistently higher than the DFT values, if we use them we are going to end up with positive values for small molecules and much smaller energies for large molecules.  A few quick calculations of the formation energy. 

Methane formation energy
DFT self energy: -1656.8639 kJ/mol
regressed self energy: 27.0328 kJ/mol

Ammonia formation energy:
DFT self energy: -1158 kJ/mol
regressed self energy: 91 kJ/mol

water: 
DFT self energy: -891 kJ/mol
regressed self energy: 74.55 kJ/mol

acetylene: 
DFT: -1612 kJ/mol
regressed: 123 kJ/mol

CCC(=O)C=O
DFT: -4922 KJ/mol
regressed: -60 KJ/mol

The larger question: does this actually matter if we can just add the same constant self energy values back on later?

Reply:
I believe this doesn't matter!